packages/dds/matrix/src/test/matrix.fuzz.spec.ts

```diff
@@ -89,18 +91,37 @@ class TypedMatrixFactory extends SharedMatrixFactory {
 // This type gets used a lot as the state object of the suite; shorthand it here.
 type State = DDSFuzzTestState<TypedMatrixFactory>;
 
-function assertMatricesAreEquivalent<T>(a: SharedMatrix<T>, b: SharedMatrix<T>) {
+async function assertMatricesAreEquivalent<T>(a: SharedMatrix<T>, b: SharedMatrix<T>) {
 	assert.equal(a.colCount, b.colCount, `${a.id} and ${b.id} have different number of columns.`);
 	assert.equal(a.rowCount, b.rowCount, `${a.id} and ${b.id} have different number of rows.`);
 	for (let row = 0; row < a.rowCount; row++) {
 		for (let col = 0; col < a.colCount; col++) {
 			const aVal = a.getCell(row, col);
 			const bVal = b.getCell(row, col);
-			assert.equal(
-				aVal,
-				bVal,
-				`${a.id} and ${b.id} differ at (${row}, ${col}): ${aVal} vs ${bVal}`,
-			);
+			if (
+				aVal !== null &&
+				typeof aVal === "object" &&
+				bVal !== null &&
+				typeof bVal === "object"
+			) {
+				const aObj: FluidObject<IFluidHandle> = aVal as FluidObject<IFluidHandle>;
+				const bObj: FluidObject<IFluidHandle> = bVal as FluidObject<IFluidHandle>;
+				const aHandle = aObj.IFluidHandle ? aObj.IFluidHandle?.get() : aObj;
+				const bHandle = bObj.IFluidHandle ? bObj.IFluidHandle?.get() : bObj;
```

Comment:
if (isObject(aVal)) {
			const aObj: FluidObject<IFluidHandle> = aVal;
			assert(
				isObject(bVal),
				`${a.id} and ${b.id} differ at ${key}: a is an object, b is not`,
			);
			const bObj: FluidObject<IFluidHandle> = bVal;
			const aHandle = aObj.IFluidHandle ? await aObj.IFluidHandle?.get() : aObj;
			const bHandle = bObj.IFluidHandle ? await bObj.IFluidHandle?.get() : bObj;

Reply:
this pattern can be reused in map an dir as well